Bottom: d384aecb28a66fa211b45dd8fb749e266e7b45b6
Top:    a8b5702504c5483dc31aa8da5668b7a44ac1af5d
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-15 00:16:38 +0000

Have new and optimised rats start at the center of pads


---

diff --git a/src/netlist.c b/src/netlist.c
index d9592b0..7cb923e 100644
--- a/src/netlist.c
+++ b/src/netlist.c
@@ -137,8 +137,10 @@ pin_name_to_xy (LibraryEntryType * pin, int *x, int *y)
       *y = ((PinType *) (conn.ptr2))->Y;
       return 0;
     case PAD_TYPE:
-      *x = ((PadType *) (conn.ptr2))->Point1.X;
-      *y = ((PadType *) (conn.ptr2))->Point1.Y;
+      *x = (((PadType *) (conn.ptr2))->Point1.X +
+            ((PadType *) (conn.ptr2))->Point2.X) / 2;
+      *y = (((PadType *) (conn.ptr2))->Point1.Y +
+            ((PadType *) (conn.ptr2))->Point2.Y) / 2;
       return 0;
     }
   return 1;
diff --git a/src/rats.c b/src/rats.c
index 35047b5..85a3f93 100644
--- a/src/rats.c
+++ b/src/rats.c
@@ -139,16 +139,8 @@ FindPad (char *ElementName, char *PinNum, ConnectionType * conn, Boolean Same)
 	    conn->ptr2 = &element->Pad[i];
 	    conn->group =
 	      TEST_FLAG (ONSOLDERFLAG, &element->Pad[i]) ? SLayer : CLayer;
-	    if (TEST_FLAG (EDGE2FLAG, &element->Pad[i]))
-	      {
-		conn->X = element->Pad[i].Point2.X;
-		conn->Y = element->Pad[i].Point2.Y;
-	      }
-	    else
-	      {
-		conn->X = element->Pad[i].Point1.X;
-		conn->Y = element->Pad[i].Point1.Y;
-	      }
+	      conn->X = (element->Pad[i].Point1.X + element->Pad[i].Point2.X) / 2;
+	      conn->Y = (element->Pad[i].Point1.Y + element->Pad[i].Point2.Y) / 2;
 	    break;
 	  }
       if (i == element->PadN)
